### ISSUE TITLE
`unlint-non-existing-defect` generates warnings, even though it was disabled

### ISSUE THREAD

**Issue by h1alexbel** [objectionary/lints#699]
This is what we do:

```java
new Program(pkg).without("unlint-non-existing-defect").defects();
```

However, we still receiving this:

```text
foo:11 Unlinting rule 'redundant-object:18' doesn't make sense, since there are no defects with it (unlint-non-existing-defect (Single))
```

I believe we need to fix it

**Comment by bibonix** [comment]
@yegor256 I'm planning to work on this. I'll trace through how `Source.without(...)` excludes lints and check why `unlint-non-existing-defect` slips past the filter.

**Comment by bibonix** [comment]
Submitted #852 with a fix and a regression test.

The bug is in `MonoWithout`: it filters `MonoLints` with `WithoutLints` before passing them into `PkMono`, but `PkMono` then unconditionally appends a fresh `LtUnlintNonExistingDefect` to the resulting list — that addition bypasses the user's exclusion list, so `without("unlint-non-existing-defect")` had no effect on it. Wrapping the `PkMono` result in a second `WithoutLints` drops any lint the caller asked to exclude, including ones added inside `PkMono`.

**Comment by volodya-lombrozo** [comment]
Resolved by #852.
